{"name": "Добавление товара в корзину", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element is not clickable at point (351, 991)\n  (Session info: chrome=127.0.6533.88)\nStacktrace:\n#0 0x61b772ccf6ba <unknown>\n#1 0x61b77299f730 <unknown>\n#2 0x61b7729f5a96 <unknown>\n#3 0x61b7729f39ad <unknown>\n#4 0x61b7729f1447 <unknown>\n#5 0x61b7729f030e <unknown>\n#6 0x61b7729e3d66 <unknown>\n#7 0x61b772a11a52 <unknown>\n#8 0x61b7729e3719 <unknown>\n#9 0x61b772a11c1e <unknown>\n#10 0x61b772a301ca <unknown>\n#11 0x61b772a117f3 <unknown>\n#12 0x61b7729e1ec9 <unknown>\n#13 0x61b7729e291e <unknown>\n#14 0x61b772c959eb <unknown>\n#15 0x61b772c99972 <unknown>\n#16 0x61b772c82e15 <unknown>\n#17 0x61b772c9a502 <unknown>\n#18 0x61b772c67d2f <unknown>\n#19 0x61b772cbe578 <unknown>\n#20 0x61b772cbe750 <unknown>\n#21 0x61b772cce48c <unknown>\n#22 0x7dc8950a0ac3 <unknown>", "trace": "browser = <selenium.webdriver.remote.webdriver.WebDriver (session=\"0f144b93d33aaf1ca59636ab0f5639fa\")>, base_url = 'http://10.0.2.15:8081'\n\n    @allure.title(\"Добавление товара в корзину\")\n    def test_add_to_cart(browser, base_url):\n        home_page = HomePage(browser)\n        home_page.open(base_url)\n        home_page.scroll_down()\n>       home_page.add_product_to_cart()\n\ntests/test_start.py:61: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\npage_objects/home_page.py:25: in add_product_to_cart\n    self.get_element(self.LISTING_CART_BUTTON).click()\n../../../.local/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:94: in click\n    self._execute(Command.CLICK_ELEMENT)\n../../../.local/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py:395: in _execute\n    return self._parent.execute(command, params)\n../../../.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x717ae3524610>, response = {'status': 400, 'value': '{\"value\":{\"error\":\"element click intercepted\",\"message\":\"element click intercepted: Element ...\\\\n#20 0x61b772cbe750 \\\\u003Cunknown>\\\\n#21 0x61b772cce48c \\\\u003Cunknown>\\\\n#22 0x7dc8950a0ac3 \\\\u003Cunknown>\\\\n\"}}'}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n            raise exception_class(value)\n        if message == \"\" and \"message\" in value:\n            message = value[\"message\"]\n    \n        screen = None  # type: ignore[assignment]\n        if \"screen\" in value:\n            screen = value[\"screen\"]\n    \n        stacktrace = None\n        st_value = value.get(\"stackTrace\") or value.get(\"stacktrace\")\n        if st_value:\n            if isinstance(st_value, str):\n                stacktrace = st_value.split(\"\\n\")\n            else:\n                stacktrace = []\n                try:\n                    for frame in st_value:\n                        line = frame.get(\"lineNumber\", \"\")\n                        file = frame.get(\"fileName\", \"<anonymous>\")\n                        if line:\n                            file = f\"{file}:{line}\"\n                        meth = frame.get(\"methodName\", \"<anonymous>\")\n                        if \"className\" in frame:\n                            meth = f\"{frame['className']}.{meth}\"\n                        msg = \"    at %s (%s)\"\n                        msg = msg % (meth, file)\n                        stacktrace.append(msg)\n                except TypeError:\n                    pass\n        if exception_class == UnexpectedAlertPresentException:\n            alert_text = None\n            if \"data\" in value:\n                alert_text = value[\"data\"].get(\"text\")\n            elif \"alert\" in value:\n                alert_text = value[\"alert\"].get(\"text\")\n            raise exception_class(message, screen, stacktrace, alert_text)  # type: ignore[call-arg]  # mypy is not smart enough here\n>       raise exception_class(message, screen, stacktrace)\nE       selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element is not clickable at point (351, 991)\nE         (Session info: chrome=127.0.6533.88)\nE       Stacktrace:\nE       #0 0x61b772ccf6ba <unknown>\nE       #1 0x61b77299f730 <unknown>\nE       #2 0x61b7729f5a96 <unknown>\nE       #3 0x61b7729f39ad <unknown>\nE       #4 0x61b7729f1447 <unknown>\nE       #5 0x61b7729f030e <unknown>\nE       #6 0x61b7729e3d66 <unknown>\nE       #7 0x61b772a11a52 <unknown>\nE       #8 0x61b7729e3719 <unknown>\nE       #9 0x61b772a11c1e <unknown>\nE       #10 0x61b772a301ca <unknown>\nE       #11 0x61b772a117f3 <unknown>\nE       #12 0x61b7729e1ec9 <unknown>\nE       #13 0x61b7729e291e <unknown>\nE       #14 0x61b772c959eb <unknown>\nE       #15 0x61b772c99972 <unknown>\nE       #16 0x61b772c82e15 <unknown>\nE       #17 0x61b772c9a502 <unknown>\nE       #18 0x61b772c67d2f <unknown>\nE       #19 0x61b772cbe578 <unknown>\nE       #20 0x61b772cbe750 <unknown>\nE       #21 0x61b772cce48c <unknown>\nE       #22 0x7dc8950a0ac3 <unknown>\n\n../../../.local/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py:229: ElementClickInterceptedException"}, "steps": [{"name": "Открываю http://10.0.2.15:8081", "status": "passed", "start": 1723384648510, "stop": 1723384649011}, {"name": "Скроллю страницу вниз", "status": "passed", "start": 1723384649011, "stop": 1723384649082}, {"name": "Добавляю товар в корзину", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.ElementClickInterceptedException: Message: element click intercepted: Element is not clickable at point (351, 991)\n  (Session info: chrome=127.0.6533.88)\nStacktrace:\n#0 0x61b772ccf6ba <unknown>\n#1 0x61b77299f730 <unknown>\n#2 0x61b7729f5a96 <unknown>\n#3 0x61b7729f39ad <unknown>\n#4 0x61b7729f1447 <unknown>\n#5 0x61b7729f030e <unknown>\n#6 0x61b7729e3d66 <unknown>\n#7 0x61b772a11a52 <unknown>\n#8 0x61b7729e3719 <unknown>\n#9 0x61b772a11c1e <unknown>\n#10 0x61b772a301ca <unknown>\n#11 0x61b772a117f3 <unknown>\n#12 0x61b7729e1ec9 <unknown>\n#13 0x61b7729e291e <unknown>\n#14 0x61b772c959eb <unknown>\n#15 0x61b772c99972 <unknown>\n#16 0x61b772c82e15 <unknown>\n#17 0x61b772c9a502 <unknown>\n#18 0x61b772c67d2f <unknown>\n#19 0x61b772cbe578 <unknown>\n#20 0x61b772cbe750 <unknown>\n#21 0x61b772cce48c <unknown>\n#22 0x7dc8950a0ac3 <unknown>\n\n", "trace": "  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n  File \"/home/fuenukkusu2501/Develop/tmp/OTUS_selenium_UI/page_objects/home_page.py\", line 25, in add_product_to_cart\n    self.get_element(self.LISTING_CART_BUTTON).click()\n  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py\", line 94, in click\n    self._execute(Command.CLICK_ELEMENT)\n  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/selenium/webdriver/remote/webelement.py\", line 395, in _execute\n    return self._parent.execute(command, params)\n  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/home/fuenukkusu2501/.local/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py\", line 229, in check_response\n    raise exception_class(message, screen, stacktrace)\n"}, "start": 1723384649082, "stop": 1723384650304}], "start": 1723384648510, "stop": 1723384650309, "uuid": "d935b8d0-827e-4e39-88b0-9b15ca095850", "historyId": "2987328dc021ab097458a9c10fd9e829", "testCaseId": "2987328dc021ab097458a9c10fd9e829", "fullName": "tests.test_start#test_add_to_cart", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_start"}, {"name": "host", "value": "Ubuntu"}, {"name": "thread", "value": "117058-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_start"}]}